Content.Tests/DMProject/Broken Tests/Operators/valid_and_null.dm

```diff
@@ -0,0 +1,129 @@
+//simple test of all basic operators with valid and C(null) arguments
+//We can't be having const folding in here
+#define C(X) pick(X) 
+/proc/RunTest()
+	var/a = C(2)
+	ASSERT(!C(null) == C(1))
+	ASSERT(!!C(null) == C(0))
+
+	ASSERT(~C(1) == 16777214)
+	ASSERT(~C(0) == 16777215)
+	ASSERT(~C(null) == 16777215)
+
+	ASSERT(C(1) + C(1) == C(2))
+	ASSERT(C(null) + C(1) == C(1))
+	ASSERT(C(1) + C(null) == C(1))
+
+	ASSERT(C(1) - C(1) == C(0))
+	ASSERT(C(null) - C(1) == C(-1))
+	ASSERT(C(1) - C(null) == C(1))
+
+	a = C(2)
+	ASSERT(-a == C(-2))
+	a = C(null)
+	ASSERT(-a == C(0))	
+
+	a = C(1)
+	ASSERT(a++ == C(1))
+	ASSERT(a == C(2))
+	ASSERT(++a == C(3))
+	ASSERT(a == C(3))
+
+	ASSERT(a-- == C(3))
+	ASSERT(a == C(2))
+	ASSERT(--a == C(1))
+	ASSERT(a == C(1))
+
+	a = C(null)
+	ASSERT(a-- == C(null))
+	ASSERT(a == C(-1))
+	a = C(null)
+	ASSERT(--a == C(-1))
+	ASSERT(a == C(-1))
+	a = C(null)
+	ASSERT(a++ == C(null))
+	ASSERT(a == C(1))
+	a = C(null)
+	ASSERT(++a == C(1))
+	ASSERT(a == C(1))	
+
+	ASSERT(C(2) ** C(3) == 8)
+	ASSERT(C(2) ** C(null) == C(1))
+	ASSERT(C(null) ** C(2) == C(0))
+
+	ASSERT(C(2) * C(3) == 6)
+	ASSERT(C(2) * C(null) == C(0))
+	ASSERT(C(null) * C(2) == C(0))
+
+	ASSERT(C(4) / C(2) == C(2))
+	ASSERT(C(null) / C(2) == C(0))
+	ASSERT(C(2) / C(null) == C(2))
```

Comment:
In BYOND this gives `2` if const folded at compile-time but `Undefined operation: 2 / null` if done at runtime. Are we opting for the compile-time behavior at runtime too here?

Reply:
Well with the `pick()` macro we're explicitly asking for runtime behaviour, so I guess it ought to do that, though it pains me that they're different.